etl/src/main/java/org/pih/hivmigration/etl/sql/VisitMigrator.groovy

```diff
@@ -0,0 +1,42 @@
+package org.pih.hivmigration.etl.sql
+
+class VisitMigrator extends SqlMigrator {
+
+    void migrate() {
+
+        executeMysql("Create visits from existing encounters", '''
+            -- Get visit type "Clinic or Hospital Visit"
+            SET @visit_type_id = (SELECT visit_type_id FROM visit_type WHERE uuid = 'f01c54cb-2225-471a-9cd5-d348552c337c');
+            SET @encounter_type_exclusions = '';  -- a comma-separated string like '1,2,3'
+            
+            INSERT INTO visit
+            (patient_id,   visit_type_id,  date_started,   date_stopped,   location_id,   creator,   date_created, voided, uuid)
+            SELECT  e.patient_id, @visit_type_id, e.date_started, e.date_stopped, e.location_id, e.creator, now(),     0, uuid()
+            FROM
+                (
+                    SELECT patient_id,
+                           Min(encounter_datetime) AS date_started,  -- encounter_datetime is always midnight
+                           Addtime(Max(encounter_datetime), '23:59:00') AS date_stopped,
+                           Max(location_id) as location_id,  /* Avoid using 'Unknown Location' if there is another location available */
+                           creator
+                    FROM   encounter
+                    WHERE  FIND_IN_SET(encounter_type, @encounter_type_exclusions) = 0
+                    GROUP  BY patient_id,
+                              Date(encounter_datetime)
+                ) AS e;
+            
+            -- Add visit IDs to their encounters
+            UPDATE encounter e
+                INNER JOIN visit v
+                ON e.patient_id = v.patient_id
+                    AND Date(e.encounter_datetime) = Date(v.date_started)
```

Comment:
Likely we also want to match on location here?

Reply:
It's an interesting point.  If I'm following the "insert" above does not group by location, but if there are multiple encounter locations within the same day it just picks one arbitrarily (or, more exactly, the one with the highest primary key, to make sure that "Uknown Location" is never chosen if there's another option).  So therefore we don't want to match on location here.

Assumedly, we shouldn't have multiple locations for the same patient on the same day, but @brandones probably worth running a quick query to see if this is the case?

Reply:
Fair enough that there shouldn't be a visit at more than one location on the same day, and would be good to check.  But regardless, adding the group by on location above then wouldn't hurt, would it?  And then adding that in here too.

Reply:
Hmm... this is making me go down a bit of a rabbit hole re: visit locations... will bring it up after standup today.

Reply:
Okay, scratch my previous comment... yes, I think we should try grouping by location, we just need to figure out how to handle any encounters at the Unknown Location, or any locations that aren't Visit locations.  Maybe handle that as a separate ticket if it's blocking this one too much.  @brandones  can you run quick distinct query count of all the encouter_locations in the system?

Reply:
Actually, looks like I already did the analysis of multiple encounters on the same day at different locations as part of this ticket:
https://pihemr.atlassian.net/browse/UHM-3239